api/views/exercises.py

```diff
@@ -0,0 +1,92 @@
+from flask import Blueprint, jsonify
+from sqlalchemy import asc
+from models import Exercise, Artifact, Language
+
+DEFAULT_LIMIT = 25
+
+
+def exercise_to_dict(exercise):
+    return {
+        'name': exercise.name,
+        'language': exercise.language.value,
+        'creator': exercise.creator,
+        'topics': exercise.topics,
+        'difficulty': exercise.difficulty,
+    }
+
+
+def complete_exercise_to_dict(exercise, artifact):
+    return {
+        'name': exercise.name,
+        'language': exercise.language.value,
+        'creator': exercise.creator,
+        'topics': exercise.topics,
+        'difficulty': exercise.difficulty,
+        'description': exercise.description,
+        'readme': artifact.readme,
+        'solution': artifact.solution,
+        'starting_point': artifact.starting_point,
+        'test': artifact.test,
+        'hint': artifact.hint,
+    }
+
+
+def create_exercises_blueprint(session, request):
+    exercises = Blueprint('exercises', __name__)
+
+    @exercises.route('/', methods=['GET'])
+    def get_exercises(default_limit=DEFAULT_LIMIT):
+        limit = request.args.get('page_size', default_limit)
+        offset = request.args.get('page', 0)
```

Comment:
`request.args` always contains strings, so you probably want to check if the value is a valid integer before using it. SQLAlchemy probably does a cast but you'll end with a 500.

Reply:
```python
try:
            limit = int(request.args.get('page_size', default_limit))
            offset = int(request.args.get('page', 0))
        except ValueError:
            abort(400)
```
Something like this, or would you do it in a better way (I hate using try -> except)?

Reply:
Well, since validation errors are common and you'll need it in every single route, it makes sense to extract this logic to a decorator and pair it with some helper functions:
```python
class ValidationError(Exception):
    def __init__(self, message, original, field):
        self.message = message
        self.original = original
        self.field = field

def handle_validation_error(func):
    @wraps(func)
    def wrapper(*args, **kwargs):
        try:
            return func(*args, **kwargs)
        except ValidationError as err:
            logging.warn(f"Validation Error: {err.original}")
            return make_response(jsonify(error=err.message, field=field), status=400)
    return wrapper

def extract_int_arg(request, key, default=0):
    try:
        return int(request.args.get(key, default))
    except ValueError as err:
        raise ValidationError("Not an integer", err, key)
```
Then you can just place the `handle_validation_error` decorator on your route functions, and use your helper functions to validate fields. Any errors will be converted to a nice error message.